BaikalDataGenerator/Source/cmd_line_parser.cpp

```diff
@@ -0,0 +1,74 @@
+/**********************************************************************
+Copyright (c) 2018 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+********************************************************************/
+
+#include "cmd_line_parser.h"
+#include <iostream>
+
+namespace
+{
+    constexpr char const* kHelpMessage =
+        "Baikal [-light_file name_of_the_light_config]"
+        "[-camera_file name_of_the_camera_config]"
+        "[-scene_file name_of_the_scene_config]"
+        "[-spp_file name_of_the_spp_config]"
+        "[-outpute_dir path_to_generate_data]"
+        "[-width output_width]"
+        "[-height output_height]"
+        "[-gamma enables_gamma_correction]";
+}
+
+CmdLineParser::CmdLineParser(int argc, char* argv[])
+    : m_cmd_parser(argc, argv)
+{   }
+
+bool CmdLineParser::OptionExists(const std::string& option) const
```

Comment:
This method is not required, the client only needs to know if '-help' is specified. Futhermore, the client should not know the exact option names. There should be something like `OptionHelpExists`.

Reply:
Nope, we need this option in standalone